gc/gc-base/src/main/java/org/projectnessie/gc/identify/IdentifyLiveContents.java

```diff
@@ -241,6 +243,7 @@ private ReferencesWalkResult identifyContentsForReference(
 
       LogEntryHolder holder = new LogEntryHolder();
       String lastCommitId = null;
+      String finalCommitId = null;
```

Comment:
nit: "last" and "final" are confusing prefixes when used together imo

Reply:
I tend to agree... How about `firstNonLiveCommitId` (to match the log message)?